feature/policy_forwarding/encapsulation/otg_tests/encap_gre_ipv4/README.md

```diff
@@ -0,0 +1,145 @@
+*  No packet loss when forwarding.
+*  Traffic equally load-balanced across 32 GRE destinations.
+*  Verify PF packet counters matching traffic generated.
+
+### PF-1.1.2: Verify PF GRE encapsulate action for IPv6 traffic
+Generate traffic on ATE Port 1 from IPV6-SRC2 from a random combination of 1000 source addresses to IPV6-DST.
+Use 512 bytes frame size.
+
+Verify:
+
+*  All traffic received on ATE Port 2 GRE-encapsulated.
+*  No packet loss when forwarding.
+*  Traffic equally load-balanced across 32 GRE destinations.
+*  Verify PF packet counters matching traffic generated.
+
+### PF-1.1.3: Verify PF IPV4 forward action
+Generate traffic on ATE Port 1 from sources IPV4-SRC1 to IPV4-DST.
+
+Verify:
+
+*  All traffic received on ATE Port 3.
+*  No packet loss when forwarding.
+
+### PF-1.1.4: Verify PF IPV6 forward action
+Generate traffic on ATE Port 1 from sources IPV6-SRC1 to IPV6-DST.
+
+Verify:
+
+*  All traffic received on ATE Port 3.
+*  No packet loss when forwarding.
+
+### PF-1.1.5: Verify PF GRE DSCP copy to outer header for IPv4 traffic
+Generate traffic on ATE Port 1 from IPV4-SRC1 source for every DSCP value in [0, 8, 16, 24, 32, 40, 48, 56]
+
+Verify:
+
+*  All traffic received on ATE Port 2 GRE-encapsulated.
+*  Outer GRE IPv4 header has same marking as ingress non-encapsulated IPv4 packet.
+
+### PF-1.1.6: Verify PF GRE DSCP copy to outer header for IPv6 traffic
+Generate traffic on ATE Port 1 from IPV6-SRC1 for every IPv6 TC 8-bit value [0, 32, 64, 96, 128, 160, 192, 224]
+
+Verify:
+
+*  All traffic received on ATE Port 2 GRE-encapsulated.
+*  Outer GRE IPv4 header has DSCP match to ingress IPv6 TC packet.
+
+### PF-1.1.7: Verify MTU handling during GRE encap
+* Generate traffic on ATE Port 1 from IPV4-SRC1 with frame size of 4000 with DF-bit set.
+* Generate traffic on ATE Port 1 from IPV6-SRC1 with frame size of 4000 with DF-bit set.
+
+Verify:
+
+*  DUT generates "Fragmentation Needed" message back to ATE source.
+
+## OpenConfig Path and RPC Coverage
+
+```yaml
+paths:
+    # match condition
+    /network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/ipv4/config/source-address:
+    /network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/ipv6/config/source-address:
+    # encap action
+    /network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/encapsulate-gre/targets/target/config/id:
+    /network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/encapsulate-gre/targets/target/config/source:
+    # either destination or identifying-prefix can be specified
+    /network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/encapsulate-gre/targets/target/config/destination:
+    /network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/encapsulate-gre/config/identifying-prefix:
+    # application to the interface
```

Comment:
Do you want to include any coverage for `/network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/encapsulate-gre/targets/target/config/ip-ttl`. (not strictly required if we aren't using this at the moment.